NBitcoin/Protocol/Node.cs

```diff
@@ -1117,6 +1146,12 @@ public void DisconnectAsync(string reason, Exception exception = null)
 			}
 		}
 
+		private void AssertListenerThreadIsNotCurrentThread()

```

Comment:
Minimal changes so I prefer not to have a method name change.

Reply:
I'll rename :)